selfdrive/updated.py

```diff
@@ -175,7 +175,7 @@ def finalize_update() -> None:
   shutil.copytree(OVERLAY_MERGED, FINALIZED, symlinks=True)
 
   run(["git", "reset", "--hard"], FINALIZED)
-  run(["git", "submodule", "foreach", "--recursive", "git", "reset"], FINALIZED)
+  run(["git", "submodule", "foreach", "--recursive", "git", "reset", "--hard"], FINALIZED)
```

Comment:
should probably also do this after the fetch https://github.com/commaai/openpilot/blob/b76829c72550f467447a8a5673dbe442c84dc1f0/selfdrive/updated.py#L374